[semver:major] Rewrite to odk spec.

gmemstr opened the pull request:
Marking as major due to some of the script changes.

gmemstr committed:
Rewrite to odk spec.

gmemstr committed:
Fix configuration typos.

gmemstr committed:
Add missing orbs.

gmemstr committed:
Linting

gmemstr committed:
Remove parameters from scripts.

gmemstr committed:
Add missing fis.

gmemstr committed:
Double single quotes to double quotes.

KyleTryon commented:
It doesnt appear integration tests have ran 
![image](https://user-images.githubusercontent.com/33272306/99688608-0128c280-2a54-11eb-86f8-86f88378b57b.png)

KyleTryon reviewed (changes requested):
- enable integration tests to run.
- Include the updated .github folder which has the cpeng team as the reviewers

gmemstr commented:
@KyleTryon There were no integration tests to speak of initially - the tests that did exist where just Bets tests to check that things were inlined correctly? Regardless, I'm not too familiar with Cloudforge myself so might take some time to get "real" tests going.

gmemstr committed:
Split auth into own command.

gmemstr committed:
Fix up parameters for install.

gmemstr committed:
Fix job commands.

gmemstr committed:
Remove typo.

gmemstr committed:
Debugging.

gmemstr committed:
fix typo.

gmemstr committed:
Alternative install method.

gmemstr committed:
Addition integration tests.

gmemstr committed:
apk add curl.

gmemstr committed:
~ -> $HOME

gmemstr committed:
Source bash env.

gmemstr committed:
source -> .

gmemstr committed:
debugging.

gmemstr committed:
Correcting PATH.

gmemstr committed:
Remove sudo for cimg/base.

Force remove sudo.

alpine with custom shell.

Revert.

gmemstr committed:
First iteration of pushing to IBM cloud.

Fix path to manifest.

Setup buildpack.

Add working directory in cases root is not correct.

working_dir -> path.

Remove space.

KyleTryon commented on src/jobs/blue_green.yml:
```diff
@@ -0,0 +1,94 @@
+  Execute a blue/green deploy  in a single job. Expects either build_steps
+  or workspace_path for assets to deploy.
+docker:
+  - image: circleci/node:10
```
hardcoded image on the old namespace. Recommend changing with image parameter and defaulting to cimg

KyleTryon commented on src/examples/push_existing_assets.yml:
```diff
@@ -0,0 +1,24 @@
+description: Deploy with artifacts created in previous CircleCI jobs
+
+usage:
```
Example appears incomplete. It appears there should maybe be a step in the build job to persist to the workspace used by the CF job. Would also recommend re-ordering the example so that version is at the top, followed by the imported orbs. some of the other examples have the same ordering issue.

KyleTryon commented on src/examples/blue_green_deploy.yml:
```diff
@@ -0,0 +1,23 @@
+              - run: echo 'you can have more, too'
+            context: your-context
+            domain: your-domain
+            manifest: null
```
Remove all "nulls" from examples. Replace with example values or remove.

KyleTryon commented on src/examples/build_and_push.yml:
```diff
@@ -0,0 +1,18 @@
+            build_steps:
+              - run: echo 'your build steps'
+              - run: echo 'you can have more, too'
+            manifest: null
```
nulls

KyleTryon commented on src/examples/build_and_push.yml:
```diff
@@ -0,0 +1,18 @@
+usage:
+  orbs:
+    cloudfoundry: circleci/cloudfoundry@x.y
+  version: 2.1
```
version ordering

KyleTryon commented on src/jobs/dark_deploy.yml:
```diff
@@ -0,0 +1,82 @@
+  Execute a dark (blue) deploy  in a single job. Expects either build_steps
+  or workspace_path for assets to deploy.
+docker:
+  - image: circleci/node:10
```
hardcoded image

KyleTryon commented on src/jobs/live_deploy.yml:
```diff
@@ -0,0 +1,47 @@
+description: Execute final URL remap and application name cleanup.
+docker:
+  - image: circleci/node:10
```
hardcoded image

KyleTryon commented on src/jobs/push.yml:
```diff
@@ -0,0 +1,62 @@
+  Execute a simple push in a single job. Expects either build_steps or
+  workspace_path for assets to deploy.
+docker:
+  - image: circleci/node:10
```
hardcoded image

gmemstr committed:
Address review comments.

KyleTryon commented on src/jobs/blue_green.yml:
```diff
@@ -1,9 +1,15 @@
+  image:
+    description: Docker image in which to run commands.
+    type: string
+    default: cimg/base:latest
```
There is no latest tag. Use Stable instead.

gmemstr replied:
My bad - muscle memory

gmemstr committed:
Latest -> Stable.
